### PR TITLE
Adds support to configure files as OSDs

### PR DESCRIPTION
**Pull request by UtkarshBhatthere**
Adds a `loop-spec` parameter to the `add-osd` action to create and use files as OSDs.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ pyroute2
 netifaces
 jsonschema
 jinja2
-git+https://opendev.org/openstack/charm-ops-sunbeam#egg=ops_sunbeam
+git+https://opendev.org/openstack/sunbeam-charms/#egg=ops-sunbeam&subdirectory=ops-sunbeam
```
Comment (lmlg) [inline requirements.txt:7]: What is this change for?

File: src/microceph.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        logger.debug(f'Running command {" ".join(cmd)}')
+        process = subprocess.run(cmd, capture_output=True, text=True, check=True, timeout=180)
+        logger.debug(f"Command finished. stdout={process.stdout}, " f"stderr={process.stderr}")
+    except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as e:
+        logger.warning(e.stderr)
+        error_disk_already_exists = (
+            "Failed adding new disk: Failed to record disk: Failed to create "
+            '"disks" entry: UNIQUE constraint failed: disks.member_id, disks.path'
+        )
+        if error_disk_already_exists not in e.stderr:
```
Comment (lmlg) [inline src/microceph.py:24]: If the disk is present, the user should at least be notified. On the other hand, why is it that this isn't considered an error, anyway? (I know this code was present before, but if it needs changing, we should do it).

File: requirements.txt (hunk 1 above)
Comment (UtkarshBhatthere) [reply requirements.txt:7]: The charm-ops-sunbeam repository is retired and moved to the sunbeam-charms repository.

File: src/microceph.py (hunk 2 above)
Comment (UtkarshBhatthere) [reply src/microceph.py]: ack, have removed the section where catch and add logic, the exception now rises directly to charm.py caller.

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -57,7 +58,8 @@ def test_all_relations(self):
             timeout=180,
         )
 
-    def test_add_osds_action(self):
+    @patch.object(microceph, "subprocess")
+    def test_add_osds_action(self, subprocess):
```
Comment (sabaini) [inline tests/unit/test_charm.py:62]: Suggest to add a test exercising the loop-spec param as well

**Review by sabaini — CHANGES_REQUESTED** [review]
Hi @UtkarshBhatthere thanks for this!
Suggest to add a unit test and some minor notes -- all inline

**Pull request closed without merging**